internal/repository/local_repository.go

```diff
@@ -0,0 +1,36 @@
+package repository
+
+import (
+	"errors"
+	"sync"
+)
+
+type LocalRepository struct {
+	data map[string]string
+	mu   sync.RWMutex
+}
+
+func NewLocalRepository() *LocalRepository {
+	return &LocalRepository{
+		data: make(map[string]string),
+	}
+}
+
+func (repo *LocalRepository) Save(originalURL string, shortedURL string) error {
+	repo.mu.Lock()
+	defer repo.mu.Unlock()
+	repo.data[shortedURL] = originalURL
+	return nil
+}
+
+var ErrorRecordNotFound = errors.New("record not found")
```

Comment:
Ошибки, которые создаются как глобальные переменные, принято называть с префиксом Err

```suggestion
var ErrRecordNotFound = errors.New("record not found")
```